File: 5 - JAVA e PHP (POO)/PHP/Aula15_Exercicio_Parte_02/Gafanhoto.php
<?php

/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/**
 * Description of Gafanhoto
 *
 * @author ronan
 */

require_once 'Pessoa.php';

class Gafanhoto extends Pessoa{
    //Atributos
    private $login;
    private $totalAssistido;
    
    //Constutor
    public function __construct($nome,$idade,$sexo,$login){
        parent::__construct($nome,$idade,$sexo); //Chamada ao Construtor de Pessoa
        $this->setLogin($login);
        $this->setTotalAssistido(0);
    }
    
    //Getters and Setters
    function getLogin() {
        return $this->login;
    }

    function getTotalAssistido() {
        return $this->totalAssistido;
    }

    function setLogin($login) {
        $this->login = $login;
    }

    function setTotalAssistido($totalAssistido) {
        $this->totalAssistido = $totalAssistido;
    }

    //Método Específico
    
    public function apresentarGafanhoto(){
        echo "<p> ========= Gafanhoto ========= </p>";
        echo "Nome: {$this->getNome()}   </br>";
        echo "Idade: {$this->getIdade()} </br>";
        echo "Sexo: {$this->getSexo()}   </br>";
        echo "Login:{$this->getLogin()}  </br>";
        echo "Total Assistido: {$this->getTotalAssistido()} </br>";
        echo "Experiência: {$this->getExperiencia()}</br>";
    }
    
    public function viuMaisUm(){
        $this->setTotalAssistido($this->getTotalAssistido() + 1);
    }
    
    //Método Abstrato
    public function ganharExperiencia() {
        $this->setExperiencia($this->getExperiencia() + 1);
    }
}

File: 5 - JAVA e PHP (POO)/JAVA/Aula15_Exercicio_Parte_02_Youtube/Youtube_Parte_02/src/youtube_parte_02/Gafanhoto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package youtube_parte_02;

/**
 *
 * @author ronan
 */
public class Gafanhoto extends Pessoa{
    //Atributos
    private String login;
    private int totalAssistido;
    
    //Construtor
    public Gafanhoto(String nome,int idade,char sexo,String login){
        super(nome,idade,sexo); //Chamada ao construtor de Pessoa
        this.setLogin(login);
        this.setTotalAssistido(0);
    }
    
    //Getters and Setters

    public String getLogin() {
        return login;
    }

    public void setLogin(String login) {
        this.login = login;
    }

    public int getTotalAssistido() {
        return totalAssistido;
    }

    public void setTotalAssistido(int totalAssistido) {
        this.totalAssistido = totalAssistido;
    }
    
    
    //Método Específico
    
    public void viuMaisUm(){
        this.setTotalAssistido(this.getTotalAssistido() + 1);
    }
    
    //Método Sobreposto
    
    @Override
    protected void apresentarGafanhoto() {
        System.out.println("========== Gafanhoto ===========");
        System.out.println("Nome: "+this.getNome());
        System.out.println("Idade: "+this.getIdade());
        System.out.println("Sexo: "+this.getSexo());
        System.out.println("Login: "+this.getLogin());
        System.out.println("Nº de vídeos assistidos: "+this.getTotalAssistido());
        System.out.println("Experiência: "+this.getExperiencia()+"\n");
    }
    
    @Override
    protected void ganharExperiencia() {
        this.setExperiencia(this.getExperiencia() + 1);
    }

    
    
}
